tests/unit/test-extension-runners/test.chromium.js

```diff
@@ -357,4 +368,58 @@ describe('util/extension-runners/chromium', async () => {
     await runnerInstance.exit();
   });
 
+  describe('reloadAllExtensions', () => {
+    let runnerInstance;
+    let wsClient;
+
+    beforeEach(async () => {
+      const {params} = prepareExtensionRunnerParams();
+      runnerInstance = new ChromiumExtensionRunner(params);
+      await runnerInstance.run();
+    });
+
+    const connectClient = async () => {
+      const wssInfo = runnerInstance.wss.address();
+      const wsURL = `ws://${wssInfo.address}:${wssInfo.port}`;
+      wsClient = new WebSocket(wsURL);
+      await new Promise((resolve) => wsClient.on('open', resolve));
+    }
+
+    afterEach(async () => {
+      wsClient.close();
+      await runnerInstance.exit();
+    });
+
+    it('does not resolve before complete message from client', async () => {
+      await connectClient();
+      await runnerInstance.reloadAllExtensions();
+      assert.fail();
```

Comment:
First, to comment on your code snippet:
Never use 

```
try {
  await somePromise.catch(handleError);
} catch (e) {
  handleError(e);
}
```

... because the first `.catch` will ensure that the promise never rejects.

Instead, do

```
try {
  await somePromise;
} catch (e) {
  handleError(e);
}
```

And if you are expecting a test, then you shouldn't use the above pattern. Because if `somePromise` were to resolve successfully, then `handleError` is not called. The code should cause a test failure if the error is not thrown, and otherwise (if an error is thrown), check the error and proceed to the next part of the test.

Secondly, it seems that you are attempting to await a promise, and you want to verify that it doesn't resolve until an explicit message has been sent. That logic is flawed, because the test would then wait forever.
Instead of waiting forever, what you should do is to have the following logic:

1. Initialize flag, e.g. to false.
2. create client and register message listener on client.
3. await `reloadAllExtensions()`.
4. Set flag to true.

The message listener should then wait a little bit (e.g. 333ms), assert that the flag is still false (i.e. that `reloadAllExtensions` has not returned yet), and then return a message.